src/stream/node/source.rs

```diff
@@ -129,31 +129,37 @@ where
                 }
                 Ok(Poll::Pending) => {
                     // nothing to collect, reschedule...
-                    return Ok(());
+                    return Ok(counter);
                 }
                 Ok(Poll::Done) => {
                     // signal end..
                     self.ended = true;
-                    return Ok(());
+                    return Ok(counter);
                 }
                 Err(error) => {
                     #[cfg(feature = "metrics")]
-                    increment_counter!(format!("{}_{}", &self.descriptor, "error_counter"),);
-                    return self.handle_source_error(error);
+                    increment_counter!("error_counter", "source" => self.descriptor.clone());
+
+                    return self.handle_source_error(error, error_counter);
```

Comment:
As discussed. no need to have counter variable within the ``handle_source_error`` function. 
Add something like this where we simply bump the counter if the error is acceptable, but otherwise return Err if it is seen as critical.
```rust
match self.handle_source_error(error) {
    Ok(_) => {
        counter += 1; // Acceptable Error, just increase counter
    }
    Err(err) {
        return Err(err); // Critical error 
    }
}
```